testing-daml-models/daml-scripts/2_first_script.md

```diff
@@ -0,0 +1,45 @@
+Next to a fixed `operator` party we want to allocate a list of parties given as an argument to the
+`initialize` script. We want them to follow the operator and the operator should follow them back.
+As before, we will be using the `allocatePartyWithHint` function. Replace the initialize script
+with:
+
+<pre class="file" data-target="clipboard">
+initialize: [Text] -> Script ()
+initialize initialParties = do
+    operator <- allocatePartyWithHint "operator" (PartyIdHint "operator")
+    ps <- forA initialParties $ \partyStr -> do
+      allocatePartyWithHint partyStr (PartyIdHint partyStr)
+    debug "done!"
+</pre>
+
+Easy! Scripts are just like usual DAML functions and follow the same syntax to declare arguments and
+type signatures.
+
+1. Now we get the other party names from the `initialParties` argument to the script. 
+1. `forA: (Applicative m) => [a] -> (a -> m b) -> m [b]` starts a `for loop` that you can
+use to map anything that returns an `Applicative` (such as `Script`) over an input list and returns
+the collected results. In fact, `forA` is just `mapA` with it's arguments flipped, but it's often a
+bit more convenient to write when your lambda spans several lines.
+
+To issue ledger updates, you can use the `createCmd` and `exerciseCmd` commands. These are the
+equivalents for scripts to the usual `create` and `exercise` commands you use in updates. Let's
+create `User` contracts for all allocated parties and have the `initialParties` follow the
+`operator`.
+
+Paste the following before the final `debug` line:
+<pre class="file" data-target="clipboard">
+    submit operator $ createCmd User with username = operator, following = ps
+    forA_ ps $ \p -> do
+      h <- submit p $ createCmd User with username = p, following = []
+      submit p $ exerciseCmd h  Follow with userToFollow = operator
+</pre>
+
```

Comment:
"When pasting the above code make sure it has the same indentation as `ps <-...``

![image](https://user-images.githubusercontent.com/56260209/81649028-f7346000-942f-11ea-8a1e-fbdebac29c1d.png)
"

Maybe add something like this as indentation in DAML matters.

Reply:
When going through it the second time I got an error

![image](https://user-images.githubusercontent.com/56260209/81650261-e8e74380-9431-11ea-9efc-130ce3c0c17f.png)

Reply:
I'll see if I can reproduce.